.github/workflows/lint.yml

```diff
@@ -0,0 +1,26 @@
+---
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
```

Comment:
If I understand correctly this will trigger on *any* [`pull_request` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) on such a branch, including things like `ready_for_review` and `closed`.

We might want to use the same as in the [Geostore](https://github.com/linz/geostore/blob/50997954ef9f9f9686f111b12659cbbfe32d9bd8/.github/workflows/test.yml#L2-L4):

```yaml
on:
  pull_request:
    types: [opened, synchronize]
```